src/helpers/ConfigHelper.ts

```diff
@@ -62,10 +62,25 @@ export class ConfigHelper {
       }
     }
   }
-
+  /**
+   * Checks if current origin matches:
+   *  - config origin exactly
+   *  - variants:
+   *    - config origin without 'www.'
+   *    - config origin with stricter protocol (we should allow http sites to initialize with https but not viceversa)
```

Comment:
I don't see the HTTP config allowing both `location` being on an HTTP and HTTPS in this method.

Reply:
I think I should have said, "weaker" not "stricter". Does that address your comment?